### PR TITLE
GitHub Actions + updated dependencies

### PR DESCRIPTION
**Pull request by paulo-ferraz-oliveira**
I wanted to replace Travis CI with GitHub Actions.

I did that.

Then I ran the analysis: `dialyzer`, `xref`, ... and got to modernize `rebar.config` a bit.

I also updated some dep.s (mostly due to the fact that analysis on more recent Erlang/OTP versions showed issues).

### CODE REVIEW

File: src/trails.erl [hunk 1]
```diff
@@ -259,7 +259,7 @@ servers() ->
 
 -spec host_matches(ranch:ref()) -> [route_match()].
 host_matches(ServerRef) ->
-  [Opts] = lists:flatten(ets:match(ranch_server, {{opts, ServerRef}, '$1'})),
+  [Opts] = lists:flatten(ets:match(ranch_server, {{proto_opts, ServerRef}, '$1'})),
```
Comment (paulo-ferraz-oliveira) [inline src/trails.erl:262]: Updated for new `ranch` version.

**Review by elbrujohalcon — APPROVED** [review]

**Comment by paulo-ferraz-oliveira** [comment]
That was fast 😄 Thanks. Now let me check whatever's breaking.

**Comment by elbrujohalcon** [comment]
As usual… the build failed _after_ I merged. 🤷

**Comment by paulo-ferraz-oliveira** [comment]
> As usual… the build failed _after_ I merged. 🤷

No biggie. That's why I prefer to start with a file there, so issues are solved before the merge. But, as usual, I won't leave you hanging. Gimme a few.

**Pull request merged**